barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -550,158 +538,160 @@ cycle_group<Builder> cycle_group<Builder>::checked_unconditional_subtract(const
+    return cycle_group(result_x, result_y, /*is_infinity=*/result_is_infinity);
 }
 
 /**
  * @brief Will evaluate ECC point subtraction over `*this` and `other`.
- *        This method uses complete addition i.e. is compatible with edge cases.
- *        Method is expensive due to needing to evaluate both an addition, a doubling,
- *        plus conditional logic to handle points at infinity.
+ * @details This method uses complete subtraction i.e. is compatible with all edge cases and is therefore expensive. To
+ * handle the possibility of x-coordinate collisions we evaluate both a subtraction (modified to avoid division by zero)
+ * and a doubling, then conditionally assign the result.
  *
  * @tparam Builder
- * @param other
- * @return cycle_group<Builder>
+ * @param other Point to subtract
+ * @return cycle_group<Builder> Result of subtraction
  */
 template <typename Builder> cycle_group<Builder> cycle_group<Builder>::operator-(const cycle_group& other) const
 {
-    if (other._is_infinity.is_constant() && other._is_infinity.get_value()) {
-        return *this;
-    }
-    if (this->_is_infinity.is_constant() && this->_is_infinity.get_value()) {
+    // If lhs is constant point at infinity, return -rhs
+    if (this->is_constant_point_at_infinity()) {
         return -other;
     }
+    // If rhs is constant point at infinity, return the lhs
+    if (other.is_constant_point_at_infinity()) {
+        return *this;
+    }
+
+    Builder* context = get_context(other);
 
     const bool_t x_coordinates_match = (x == other.x);
     const bool_t y_coordinates_match = (y == other.y);
-    const bool_t double_predicate = (x_coordinates_match && !y_coordinates_match).normalize();
-    const bool_t infinity_predicate = (x_coordinates_match && y_coordinates_match).normalize();
-    if (!infinity_predicate.is_constant()) {
-        infinity_predicate.get_context()->update_used_witnesses(infinity_predicate.get_normalized_witness_index());
-    }
+
     const field_t x1 = x;
     const field_t y1 = y;
     const field_t x2 = other.x;
     const field_t y2 = other.y;
+
+    // Execute point addition with modified lambda = (-y2 - y1)/(x2 - x1 + x_coordinates_match) to avoid the possibility
+    // of division by zero.
     const field_t x_diff = x2.add_two(-x1, x_coordinates_match);
-    // Computes lambda = (-y2-y1)/x_diff, using the fact that x_diff is never 0
+    // Compute lambda in one of two ways depending on whether either numerator or denominator is constant or not
     field_t lambda;
     if ((y1.is_constant() && y2.is_constant()) || x_diff.is_constant()) {
         lambda = (-y2 - y1).divide_no_zero_check(x_diff);
     } else {
-        lambda =
-            field_t::from_witness(this->get_context(other), (-y2.get_value() - y1.get_value()) / x_diff.get_value());
+        // Note: branch saves one gate vs just using divide_no_zero_check because we avoid computing y2 - y1 in circuit
```

Comment:
super nit: edit the comment to:
```cpp
// Note: branch saves one gate vs just using divide_no_zero_check because we avoid computing (-y2 - y1) in circuit
```